src/query/settings/src/lib.rs

```diff
@@ -491,6 +492,38 @@ impl Settings {
         }))
     }
 
+    pub fn set_file_format_options(&self, opts: &FileFormatOptions) -> Result<()> {
```

Comment:
when we already have FileFormatOptions, should try to avoid using settings (our final goal)

Reply:
This is for compatibility, if we decide to remove the old syntax, this will be removed future.

Reply:
we should only support new syntax internally,  and  try to convert old syntax to new.

Reply:
FileFormatOptions -> setting is the reverse

Reply:
we should only support new syntax internally, and try to convert old syntax to new.

it is also ok to keep both code path for now.
but new syntax should not based on the old,  so we need InputContext::try_create_from_insert_v2